tests/ui/instrument-xray/target-not-supported.rs

```diff
@@ -1,7 +1,7 @@
 // Verifies that `-Z instrument-xray` cannot be used with unsupported targets,
 //
 //@ needs-llvm-components: x86
-//@ compile-flags: -Z instrument-xray --target x86_64-apple-darwin
+//@ compile-flags: -Z instrument-xray --target x86_64-pc-windows-msvc
```

Comment:
Question: do we have any smoke test for xray on `x86_64-apple-darwin` and `aarch64-apple-darwin`? As in:

- A smoke test that have some ability to observe xray being enabled or not, for `aarch64-apple-darwin` and `x86_64-apple-darwin` targets.
- A test check that the flag `-Z instrument-xray` is still unstable, but is no longer rejected on `x86_64-apple-darwin` and `aarch64-apple-darwin` respectively.

Reply:
I think both of these tests don't exist at the moment, I can add tests for this but there are two problems

1. It seems that only llir is tested in `tests/codegen` and no asm, which cannot be used to check whether xray is enabled, because xray affects llvm output rather than input. i need some guidance how to write tests for asm output.
2. Add a test specifically for `instrument-xray` + `x86_64-apple-darwin`/`aarch64-apple-darwin` combination seems a bit strange to me, since this is not a feature we implement, but just an upstream feature. do we need to add such a test for each supported target?

Reply:
1. There's also `tests/assembly` which does exercise assembly.
2. What I meant was just cli smoke tests to check that we properly gate the stability given the target platform. Whether the impl works or not is a separate question.